### ISSUE TITLE
JSON parse error when trying to add content instance on Moodle 3.1 / Totara 12

### ISSUE THREAD

**Issue by logan-reynolds** [h5p/h5p-moodle-plugin#258]
When attempting to add an instance of Accordian with latest H5P version 2019030700 from Moodle.org on either Moodle 3.1 or Totara 12, I'm seeing the following errors in browser console which indicate an error in parsing JSON data, and the editor fields never load (just continuously spinning Loading icon):
```
Ajax request failed h5peditor.js:236:13
Object { readyState: 4, getResponseHeader: getResponseHeader(), getAllResponseHeaders: getAllResponseHeaders(), setRequestHeader: setRequestHeader(), overrideMimeType: overrideMimeType(), statusCode: statusCode(), abort: abort(), state: state(), always: always(), then: then(), … }
h5peditor.js:237:13
parsererror h5peditor.js:238:13
SyntaxError: "JSON.parse: unexpected character at line 1 column 1 of the JSON data"

    parseJSON4jQuery

h5peditor.js:239:13
```
Note that the content types seem to install ok, the error only happens when you click the "Use" button after successful installation.
![h5p-console-error](https://user-images.githubusercontent.com/4237443/54076186-d7bb0980-4265-11e9-91ee-7ce6c5f427af.png)

**Comment by icc** [comment]
Thank you for the detailed data. 
Could you also press the 'Network' tab in the screenshot and look at the details for the failing request? What does the Response look like?

**Comment by logan-reynolds** [comment]
The response which appears to have invalid JSON data is to the following relative URL, although this request does return with a status 200 OK:

/mod/hvp/ajax.php?contextId=156&token=736a9e998b8a7&action=libraries&machineName=H5P.Accordion&majorVersion=1&minorVersion=0

Looks like some code is unintentionally being returned alongside the JSON data per attached screenshot below:

![malformed-h5p-json-response](https://user-images.githubusercontent.com/4237443/54170108-fe7e6900-443a-11e9-8d84-c04a99888463.png)

**Comment by icc** [comment]
Hi, it appears you have `$DB->set_debug(true);` somewhere, perhaps the config.php?
This will break AJAX calls.

**Comment by mz83ude** [comment]
Same problem with Moodle 3.3
The `$DB->set_debug(true);` is in */hvp/classes/editor_framework.php:334 (per default?)

Any suggestions?

**Comment by icc** [comment]
You are correct. This appears to have been a mistake on our side. It has been fixed in d068fc6.
We'll release a updated version as soon as possible!

**Comment by thomasmars** [comment]
Version 1.17 has now been officially released. Please update.
